src/main/python/pybuilder/plugins/python/distutils_plugin.py

```diff
@@ -312,32 +310,142 @@ def build_scripts_string(project):
     if scripts_dir:
         scripts = list(map(lambda s: os.path.join(scripts_dir, s), scripts))
 
-    return str(scripts)
+    if len(scripts) > 0:
+        return build_string_from_array(scripts)
+    else:
+        return '[]'
 
 
 def build_data_files_string(project):
+    indent = 8
+    """
+    data_files = [
+      ('bin', ['foo','bar','hhrm'])
+    ]
+    """
     data_files = project.files_to_install
-
     if not len(data_files):
-        return ""
+        return '[]'
+
+    returnString = "[\n"
+
+    for dataType, dataFiles in data_files:
+        returnString += (" " * (indent+4)) + "('%s', ['" % dataType
+        returnString += "', '".join(dataFiles)
+        returnString += "']),\n"
 
-    return "data_files = %s," % str(data_files)
+    returnString = returnString[:-2] + "\n"
+    returnString += " " * indent + "]"
+    return returnString
+    return build_string_from_array(data_files)
 
 
 def build_package_data_string(project):
+    indent = 8
+
+    sorted_keys = sorted(project.package_data.keys())
+
     package_data = project.package_data
     if package_data == {}:
-        return ""
-    package_data_string = "package_data = {"
+        return "{}"
+
+    returnString = "{\n"
+
+    for pkgType in sorted_keys:
+        returnString += " " * (indent+4)
+        returnString += "'%s': " % pkgType
+        returnString += "['"
+        returnString += "', '".join(package_data[pkgType])
+        returnString += "'],\n"
+
+    returnString = returnString[:-2] + "\n"
+    returnString += " " * indent + "}"
+
+    return returnString
+
+    return build_string_from_dict(package_data)
+
+
+def build_packages_string(project):
+    pkgs = [pkg for pkg in project.list_packages()]
+    if len(pkgs) > 0:
+        return build_string_from_array(pkgs)
+    else:
+        return '[]'
+
+
+def build_modules_string(project):
+    mods = [mod for mod in project.list_modules()]
+    if len(mods) > 0:
+        return build_string_from_array(mods)
+    else:
+        return '[]'
+
+
+def build_console_scripts_string(project):
+    console_scripts = project.get_property('distutils_console_scripts', [])
+
+    if len(console_scripts) == 0:
+        return "{}"
+
+    indent = 12
+    string = "{'console_scripts': "
+    string += build_string_from_array(console_scripts, indent)
+    string += "}"
+
+    return string
+
+
+def build_classifiers_string(project):
+    classifiers = project.get_property('distutils_classifiers', [])
+    return build_string_from_array(classifiers, indent=12)
+
+
+def build_string_from_array(arr, indent=12):
+    returnString = ""
+
+    if len(arr) == 1:
+        if not isinstance(arr[0], (frozenset, list, set, tuple)):
```

Comment:
oh nevermind, I see now what you are referring to.

I'm trying to express something that is not a list/tuple. There doesn't seem to be a straight forward way to do this since string is iterable.